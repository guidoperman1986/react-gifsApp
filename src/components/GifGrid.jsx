import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifItem } from './GifItem';

export const GifGrid = ({category}) => {

    const { images, isLoading } = useFetchGifs(category);

    return (
        <>
            <h3>{category}</h3>
            {
                !isLoading 
                    ?
                    <div className='card-grid'>
                        {
                            images.map( image => 
                                <GifItem 
                                    key={image.id} 
                                    {...image}
                                />
                            )
                        }
                    </div>
                    : <h1 style={{display: 'flex', alignSelf: 'center'}}>Loading...</h1>

            }
        </>
    )
}
