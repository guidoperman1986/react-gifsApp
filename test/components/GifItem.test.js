import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

describe('test on gifItem', () => {
    const title = 'pato';
    const url = 'http://localhost/pato.com';


    test('should match the snapshot', () => {
        const { container } = render(<GifItem title={title} url={url} />)
        expect(container).toMatchSnapshot();
    })

    test('should show image with url and the alt', () => { 
        render(<GifItem title={title} url={url} />)
        // screen.debug();

        // expect(screen.getByRole('img').src).toBe(url)
        // expect(screen.getByRole('img').alt).toBe(title)

        const { src, alt } = screen.getByRole('img');

        expect(src).toBe(url)
        expect(alt).toBe(title)
    })

    test('should show the title in the component', () => {
        render(<GifItem title={title} url={url} />)

        expect( screen.getByText(title) ).toBeTruthy();
    })
})