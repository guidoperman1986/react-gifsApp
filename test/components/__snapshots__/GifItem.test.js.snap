// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test on gifItem should match the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="pato"
      src="http://localhost/pato.com"
    />
    <p>
      pato
    </p>
  </div>
</div>
`;
